src/game_state.rs

```diff
@@ -112,11 +113,13 @@ pub struct GameState {
 }
 
 impl GameState {
-    pub fn new() -> Self {
+    pub fn new(seed: Option<u32>) -> Self {
```

Comment:
Is it ever created without a seed?

Reply:
Yes, if the `--seed` command line argument is omitted. In that case `GameState::new` will generate a random seed.